[hovanleong] iP

hovanleong opened the pull request:
# Slothing Waffler
> ["What we fear doing most is usually what we most need to do.” – Tim Ferriss](https://dansilvestre.com/productivity-quotes/)

Slothing Waffler frees your mind of having to remember things you need to do. It's,
- text-based
- easy to use
- ~~FAST~~ ***SUPER FAST*** to learn

How the name Slothing Waffler came about
1. I was talking to my friend about the iP
2. She uses the words _sloth_ and _waffling_ frequently
3. I suggested the name Waffling Slother
4. She wanted Slothing Waffler, and so Slothing Waffler was born  :shipit:

Best of all, it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, here's a random code snippet that may bring back memories:
```java
public class Main {
    public static void main(String[] args) {
        System.out.println("Hello, World!");
    }
}
```

Just simply compile and run `java Main` on your terminal.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

hovanleong committed:
Rename chatbot and main class

hovanleong committed:
Add greeting

hovanleong committed:
Tidy up main method

hovanleong committed:
Add support for echo

hovanleong committed:
Add in exit support when user inputs "bye"

hovanleong committed:
Add ability for user to add and list tasks

hovanleong committed:
Create Task.java

hovanleong committed:
Add the ability to mark tasks as done

hovanleong committed:
Create Todo.java, Deadline.java and Event.java files

hovanleong committed:
Abstract String representation of Tasks from SlothingWaffler.java to Task.java

hovanleong committed:
Add support for tracking all three types of tasks

hovanleong committed:
Create SlothingWafflerException.java to handle errors

hovanleong committed:
Add in support to handle various errors using Exceptions

hovanleong committed:
Add support for deleting tasks from the list

hovanleong committed:
Utilise the input/output redirection technique to semi-automate the testing of Slothing Waffler

hovanleong committed:
Abstract initial greeting text

hovanleong committed:
Abstract bot exit message

hovanleong committed:
Abstract printing of message after user adds new tasks

hovanleong committed:
Edit access modifiers of fields in Task.java

hovanleong committed:
Edit access modifiers of fields in Event.java

hovanleong committed:
Edit access modifiers of fields in Deadline.java

hovanleong committed:
Refactor Main.java to abstract adding of a Todo Task

hovanleong committed:
Refactor Main.java to abstract adding of Deadline Tasks and Event Tasks

hovanleong committed:
Refactor main method in SlothingWaffler.java to make it less verbose

hovanleong committed:
Change error message for thrown SlothingWafflerException

hovanleong committed:
Change error message for thrown SlothingWafflerException in main method

hovanleong committed:
Create new Storage class to read and write data to and from the hard disk

hovanleong committed:
Create new Ui class to handle Ui functionalities

hovanleong committed:
Save the tasks in the hard disk automatically whenever the task list changes

hovanleong committed:
Handle file does not exist case

hovanleong committed:
Merge branch 'branch-Level-7'

hovanleong committed:
Neaten line spacings in code for TaskList.java

hovanleong committed:
Store deadline and event dates as as LocalDateTime

hovanleong committed:
Store deadline dates as as LocalDate

hovanleong committed:
Merge branch 'branch-Level-8'

hovanleong committed:
Debug error which loads all tasks as done when bot is terminated and run again

hovanleong committed:
Create Parser class and extract Ui, Storage, Parser, TaskList classes from SlothingWaffle class

hovanleong committed:
Put all classes into slothingwaffler package

hovanleong committed:
Merge branch 'add-gradle-support'

hovanleong committed:
Change main class on build.gradle to slothingwaffler.SlothingWaffler

hovanleong committed:
Add JUnit tests to test three methods in TaskList.java and two methods in Deadline.java

hovanleong committed:
Debug computation error in number of remaining tasks left after deleting tasks

hovanleong committed:
Package app as a JAR file

hovanleong committed:
Add javadoc comments for most public methods and most public classes

hovanleong committed:
Tweak comments

hovanleong committed:
Give users a way to find a task by searching for a keyword

hovanleong committed:
Merge branch 'A-CodingStandard'

hovanleong committed:
Merge branch 'Level-9'

hovanleong committed:
Add checkstyle.xml and suppressions.xml and edit build.gradle so that coding style checks can be performed

hovanleong committed:
Correct most style errors

hovanleong committed:
Rectified all coding style errors except a handful of missing JavaDoc comments which are not mandated by project requirements

hovanleong committed:
Add in more JavaDoc comments and correct all coding style errors in classes in the test files

hovanleong committed:
Add JavaDoc comments for Parser.java file

hovanleong committed:
Merge branch 'A-CheckStyle'

hovanleong committed:
Change descriptive comment in Task class

hovanleong committed:
Add JavaDoc comments for StorageStub class

hovanleong committed:
Add more JavaDoc comments

xqtann commented on src/main/java/slothingwaffler/SlothingWaffler.java:
```diff
@@ -0,0 +1,57 @@
+package slothingwaffler;
```
Perhaps its better to split packages into sub-packages? For example `slothingwaffler.commands` and `slothingwaffler.tasks`. 🤔

xqtann commented on src/main/java/slothingwaffler/Storage.java:
```diff
@@ -0,0 +1,146 @@
+            String description = parts[2];
+            Task task = null;
+
+            switch (taskType) {
```
Great job on following coding standards for `switch case` statements! 👍

xqtann commented on src/main/java/slothingwaffler/TaskList.java:
```diff
@@ -0,0 +1,136 @@
+
+    public Task get(int i) {
+        return tasks.get(i);
+    }
```
Would it be better if JavaDocs comments are added for these methods as well? 🤔

xqtann commented on src/main/java/slothingwaffler/Task.java:
```diff
@@ -0,0 +1,56 @@
+ * This is an abstract class representing a task with a description and a status indicating whether it's done.
+ * </p>
+ */
+public abstract class Task {
```
I like how you made the `Task` class abstract, as it makes sense to not be instantiated. 👍

xqtann reviewed (commented):
LGTM. Just some nits to fix. Overall great code and coding standards were mostly followed. Well done! 😄 👍

hovanleong committed:
Implement a GUI for SlothingWaffler chatbot

hovanleong committed:
Implement automatic resizing of elements

hovanleong committed:
Debug computation error for marking and deleting of tasks

hovanleong committed:
Comment out code for text-based UI

hovanleong committed:
Merge branch 'branch-Level-10'

hovanleong committed:
Chatbot will exit automatically upon user terminating current use

hovanleong committed:
Start implementing greeting message when user launches app

hovanleong committed:
Refactor TaskListTest with appropriate test cases

hovanleong committed:
Edit JavaDoc comment in MainWindow.java

hovanleong committed:
Debug wrong outputs when user wishes to find tasks

hovanleong committed:
Prompt the chatbot to greet users upon app launch

hovanleong committed:
Change SlothingWaffler personality

weiseng18 commented on src/main/java/slothingwaffler/Deadline.java:
```diff
@@ -0,0 +1,24 @@
+package slothingwaffler;
```
great that everything is organized into a package!

you can consider having sub-packages to get more detailed with the organization

weiseng18 commented on src/main/java/slothingwaffler/Event.java:
```diff
@@ -0,0 +1,27 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + this.from.format(Task.DTF2)
```
good line break!

weiseng18 commented on src/main/java/slothingwaffler/MainWindow.java:
```diff
@@ -0,0 +1,67 @@
+
+    private SlothingWaffler slothingWaffler;
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/sleeper.png"));
```
good logical unit separation

weiseng18 commented on src/main/java/slothingwaffler/Task.java:
```diff
@@ -0,0 +1,56 @@
+    protected static final DateTimeFormatter DTF1 = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+    protected static final DateTimeFormatter DTF2 = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
+    protected static final DateTimeFormatter DTF3 = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    protected static final DateTimeFormatter DTF4 = DateTimeFormatter.ofPattern("MMM dd yyyy");
```
perhaps might be good to add a line break after this to introduce logical separation

weiseng18 reviewed (commented):
great coding standard!

hovanleong committed:
Remove unused code that became redundant

Unused code that was initially kept has now become redundant. This code,
while once considered valuable, no longer serves a purpose in the current
codebase. Keeping such code adds unnecessary complexity, making the codebase
harder to maintain and understand.

Redundant code often feels difficult to delete, especially if significant
time was invested in writing it. However, retaining it only clutters the
project. Every piece of code should add value, and if it doesn't, it's
baggage. If needed in the future, it can always be recovered from the
revision control system.

Let's remove this unused code to improve code quality, maintainability,
and readability. By doing so, we acknowledge that deleting previously written
code is a sign of refinement and continuous improvement.

Removing unused code ensures a clean and focused codebase, which is essential
for ongoing development.

hovanleong committed:
Remove unused import statements

hovanleong committed:
Add some assert statements

hovanleong committed:
Use linespace to adhere to logical structure

hovanleong committed:
Separate fields in Task class

hovanleong committed:
Merge pull request #1 from hovanleong/branch-A-Assertions

Add some assert statements

hovanleong committed:
Merge branch 'master' into branch-A-CodeQuality

hovanleong committed:
Merge pull request #2 from hovanleong/branch-A-CodeQuality

Adhere to logical structure

hovanleong committed:
Add "priority" feature for task prioritisation

The chatbot did not previously support prioritising tasks by
moving them to the top of the list, which limited users' ability to
highlight important tasks.

Let's implement a feature that allows users to move tasks to the top of
their list. This enhancement will enable users to bring critical tasks to
the forefront and manage their task list more effectively.

* Added a "priority" option for tasks.
* Updated the `TaskList` class to reorder tasks, ensuring that tasks
  moved to the top appear first.

This feature improves task management by providing a way to prioritise
important tasks and ensures they are easily visible and accessible.

hovanleong committed:
Merge branch 'branch-A-CodeQuality'

hovanleong committed:
Change logic behind prioritising tasks

hovanleong committed:
Add JUnit test for new prioritiseTask() method

hovanleong committed:
Add JavaDoc comments for prioritiseTask method in TaskList.java

hovanleong committed:
Implement sort by deadlines functionality

hovanleong committed:
Improve GUI of chatbot

hovanleong committed:
Make profile pictures circular

hovanleong committed:
Change app icon and title

hovanleong committed:
Give Slothing Waffler a more unique personality

hovanleong committed:
Add product screenshot

hovanleong committed:
Do up README.md

hovanleong committed:
Update README.md

hovanleong committed:
Add information to README.md

hovanleong committed:
Rectify formatting errors in User Guide

hovanleong committed:
Improve on User Guide

hovanleong committed:
Shorten methods in TaskList.java

hovanleong committed:
Obey SLAP Principle in TaskList.java

hovanleong committed:
Rectify errors in TaskList.java

hovanleong committed:
Remove magic literal in MainWindow.java

hovanleong committed:
Obey SLAP Principle in Storage.java
